File: java/src/test/java/mx/iteso/adapter/GoogletoWordAdapterTest.java
package mx.iteso.adapter;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.*;
import org.mockito.Mockito.*;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Matchers.eq;

import java.awt.Font;


public class GoogletoWordAdapterTest {
    GoogleToWordAdapter gToWordAdapter;
    GoogleDoc googledoc;

    @Before
    public void setUp(){
        googledoc = new GoogleDoc();
        gToWordAdapter = new GoogleToWordAdapter(googledoc);

    }

    @Test
    public void testGetFormat(){
        assertEquals("Ubuntu", gToWordAdapter.getFormat().getFont().getFontName());
    }

    @Test
    public void testGetBackground() {
        assertEquals("img.png", gToWordAdapter.getBackground().getUrl());
    }

    @Test
    public void testGetLicense() {
        assertEquals(10001, gToWordAdapter.getLicense().getNumber());
    }

    @Test
    public void testRestricEditValid() {
        assertEquals(false, gToWordAdapter.restrictEditIfLicenseIsInvalid(gToWordAdapter.getLicense()));
    }

    @Test
    public void testRestricEditInvalid() {
        googledoc.setSharingPermissions(0);
        assertEquals(true, gToWordAdapter.restrictEditIfLicenseIsInvalid(gToWordAdapter.getLicense()));
    }
}

File: python/src/unittest/python/GoogleToWordAdapter_test.py
from mockito import mock, verify, when
import unittest

from GoogleToWordAdapter import * 
from GoogleDoc import *


class GoogleToWordAdapter_tests(unittest.TestCase):

    def setUp(self):
        self.googledoc = GoogleDoc()
        self.gToWordAdapter = GoogleToWordAdapter(self.googledoc)

    def test_getFormat(self):
        self.assertEqual("Ubuntu", self.gToWordAdapter.getFormat().getFont().getName())

    def test_getBackground(self):
        self.assertEqual("img.png", self.gToWordAdapter.getBackground().getUrl());

    def test_getLicense(self):
        self.assertEqual(10001, self.gToWordAdapter.getLicense().getNumber())
    
    def test_restrict_valid(self):
        self.assertEqual(False, self.gToWordAdapter.restrictEditIfLicenseIsInvalid(self.gToWordAdapter.getLicense()))
    
    def test_restrict_invalid(self):
        self.googledoc.setSharingPermissions(0)
        self.assertEqual(True, self.gToWordAdapter.restrictEditIfLicenseIsInvalid(self.gToWordAdapter.getLicense()))